### PR TITLE
Substitute `dplyr::filter` with `base::subset` 

### PR DESCRIPTION
**Pull request by m7pr**
`base::subset()` has a bit different behavior than `dplyr::filter()` - you can see on a below code.

```r
identical(
  dplyr::filter(iris, Species == "virginica"),
  subset(iris, Species == "virginica") 
)
> [1] FALSE
```

If there were no custom `row.names()` assigned to dataframe, `dplyr::filter()` will reset rownames to `1:nrow()` where `base::subset()` preserves IDs of rows in the original dataset.

We either could look at the different level for `iris` in tests, or we need to reset rownames for iris to get tests to pass

```r
identical(
  dplyr::filter(iris, Species == "virginica"),
  subset(iris, Species == "virginica") |> `rownames<-`(1:50)
)
> [1] TRUE
```

### CODE REVIEW

File: tests/testthat/test-module_nested_tabs.R [hunk 1]
```diff
@@ -445,8 +447,9 @@ testthat::test_that("srv_teal_module: teal_module receives transformed data with
       datasets()$set_filter_state(teal_slices(teal_slice("mtcars", "cyl", selected = "6")))
       session$flushReact()
       out <- module_out()
-      testthat::expect_identical(out()[["iris"]], dplyr::filter(iris, Species == "virginica") |> head(6))
-      testthat::expect_identical(out()[["mtcars"]], dplyr::filter(mtcars, cyl == 6) |> head(6))
+      # subset keeps the old dataframe row.names where dplyr::filter used in the app resets row.names to 1:nrow()
+      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% `rownames<-`(1:50) %>% head(6))
```
Comment (averissimo) [inline tests/testthat/test-module_nested_tabs.R:451]:
```suggestion
      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% magrittr::set_rownames(1:50) %>% head(6))
```

File: tests/testthat/test-module_nested_tabs.R [hunk 2]
```diff
@@ -392,8 +393,9 @@ testthat::test_that("srv_teal_module: teal_module receives filtered data with fi
       datasets()$set_filter_state(teal_slices(teal_slice("mtcars", "cyl", selected = "6")))
       session$flushReact()
       out <- module_out()
-      testthat::expect_identical(out()[["iris"]], dplyr::filter(iris, Species == "virginica"))
-      testthat::expect_identical(out()[["mtcars"]], dplyr::filter(mtcars, cyl == 6))
+      # subset keeps the old dataframe row.names where dplyr::filter used in the app resets row.names to 1:nrow()
+      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% `rownames<-`(1:50))
```
Comment (averissimo) [inline tests/testthat/test-module_nested_tabs.R:397]:
```suggestion
      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% magrittr::set_rownames(1:50))
```

File: tests/testthat/test-module_nested_tabs.R [hunk 3]
```diff
@@ -283,7 +283,8 @@ testthat::test_that("srv_teal_module: teal_module gets data even if any transfor
       datasets()$set_filter_state(teal_slices(teal_slice("iris", "Species", selected = "virginica")))
       session$flushReact()
       out <- module_out()
-      testthat::expect_identical(out()[["iris"]], dplyr::filter(iris, Species == "virginica") |> head(6))
+      # subset keeps the old dataframe row.names where dplyr::filter used in the app resets row.names to 1:nrow()
+      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% `rownames<-`(1:50) %>% head(6))
```
Comment (averissimo) [inline tests/testthat/test-module_nested_tabs.R:287]:
```suggestion
      testthat::expect_identical(out()[["iris"]], subset(iris, Species == "virginica") %>% magrittr::set_rownames(1:50) %>% head(6))
```

**Review by averissimo — APPROVED** [review]
Cool!!

`base::subset` doesn't produce any R CMD check errors here.

Made a suggestion on `rownames<-` replacement using `{magrittr}`

File: tests/testthat/test-module_nested_tabs.R (hunk 1 above)
Comment (m7pr) [reply tests/testthat/test-module_nested_tabs.R:451]: @averissimo this is good addition, but I had a call the other day where I understood we rather remove `magrittr` from `teal` codebase

Comment (averissimo) [reply tests/testthat/test-module_nested_tabs.R:451]: If that's the choice then merge away! Other than create a function only for this, there are not many alternatives.

**Comment by m7pr** [comment]
This branch is way too conflicted. I removed `dplyr::filter` calls on the main branch in this commit 2595cd05247e58f7071f9b840430c506d45a0fdd

**Pull request closed without merging**